x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/type/DataType.java

```diff
@@ -221,6 +221,10 @@ public static boolean isPrimitive(DataType t) {
         return t != OBJECT && t != NESTED && t != UNSUPPORTED;
     }
 
+    public static boolean isObjectOrNested(DataType t) {
```

Comment:
I think this should be is**Not**ObjectOrNested 
Or isNotObjectAndNotNested? The first one is ambiguous, the second one is ugly...

Reply:
Good catch!

Another thing we could do is to just remove this method altogether - because it's only used in 2 places:

https://github.com/elastic/elasticsearch/blob/ca1b4872bf03cdb7743aa0f2e567410434966c5a/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Analyzer.java#L241

https://github.com/elastic/elasticsearch/blob/ca1b4872bf03cdb7743aa0f2e567410434966c5a/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Analyzer.java#L862

From what I have seen we cannot use the other implementation of `isPrimitive` which is why I just renamed the method:

https://github.com/elastic/elasticsearch/blob/ca1b4872bf03cdb7743aa0f2e567410434966c5a/x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/type/DataType.java#L220-L222

Reply:
I don't have a particular preference.
We could also revert the logic and use `isObjectOrNested(a.dataType) == false`, but it would make it a bit verbose probably.
Up to you, any of the above is good for me

Reply:
Done - I went with your last suggestion